todo_app/app.py

```diff
@@ -1,15 +1,46 @@
-from flask import Flask
+from todo_app.data import session_items
+from flask import Flask, render_template
+from flask.globals import request, session
 
 from todo_app.flask_config import Config
 
 app = Flask(__name__)
 app.config.from_object(Config)
 
 
-@app.route('/')
-def index():
-    return 'Hello World!'
+@app.route('/', methods=["GET"])
+def index_get():
+    items =  session_items.get_items()
+    return render_template('index.html',items=items)
 
 
+@app.route('/', methods=["POST"])
+def edit_item():
+    if "deletebutton" in request.form:
+        val = request.form.get("deletebutton")
+        print(val)
+        session_items.delete_item(val)
+    elif "Complete" in request.form:
+        val = request.form.get("Complete")
+        item = session_items.get_item(int(val))
+        item['status'] = "Completed"
+        session_items.save_item(item)
+    elif "Reset" in request.form:
+        val = request.form.get("Reset")
+        item = session_items.get_item(int(val))
+        item['status'] = "Not Started"
+        session_items.save_item(item)
+    items =  session_items.get_items()
+    return render_template('index.html',items=items)
```

Comment:
Here, we return a template directly out of a post method; this can lead to odd behaviour like if the user refreshes the page then they duplicate-add another item.
Instead, we could make use of redirects, which also have the advantage of saving us from duplicating the index route's functionality if that changes

e.g. in flask, we can use the `redirect` function. [There's more info in the flask docs](https://flask.palletsprojects.com/en/1.1.x/quickstart/), and an example from those docs:
```
@app.route('/')
def index():
    return redirect(url_for('login'))

@app.route('/login')
def login():
   # Do something
```